src/ledger/LedgerManagerImpl.cpp

```diff
@@ -855,9 +855,15 @@ LedgerManagerImpl::closeLedger(LedgerCloseData const& ledgerData)
         LedgerUpgrade lupgrade;
         try
         {
-            xdr::xdr_from_opaque(sv.upgrades[i], lupgrade);
+            if (!Upgrades::isValid(sv.upgrades[i], lupgrade,
+                                   ltx.loadHeader().current(),
+                                   Config::CURRENT_LEDGER_PROTOCOL_VERSION))
```

Comment:
As much as possible, we should not use `Config::CURRENT_LEDGER_PROTOCOL_VERSION` (and use `cfg.LEDGER_PROTOCOL_VERSION`) as some tests are testing mixed version, and we want them to behave as close as possible to "the old version" for that protocol (ie they behave like the old code)